patches/minecraft/net/minecraft/entity/player/ServerPlayerEntity.java.patch

```diff
@@ -150,6 +150,20 @@
     }
  
     public ChatVisibility func_147096_v() {
+@@ -1276,7 +_,12 @@
+ 
+    @Nullable
+    public ITextComponent func_175396_E() {
+-      return null;
++      if (!this.hastTablistName)
```

Comment:
Can't you just do a null check?

Reply:
No. As explained [here](https://github.com/noeppi-noeppi/MinecraftForge/blob/feature/tablist_name/patches/minecraft/net/minecraft/entity/player/ServerPlayerEntity.java.patch#L225) this field is null when the client should determine the tab list name itself by checking the scoreboard team. So this variable will be null in most cases and cacheing it would no longer make sense as the event would be posted every time the method is called.

Reply:
Ahh yes, moment of madness, I understand now.